Take into account timeout limit when retrying queries (#46046)
rails/rails: guides/source/configuring.md

  use_metadata_table: false
```

#### Configuring Retry Behaviour

By default, Rails will automatically reconnect to the database server and retry certain queries
if something goes wrong. Only safely retryable (idempotent) queries will be retried. The number
of retries can be specified in your the database configuration via `connection_retries`, or disabled
by setting the value to 0. The default number of retries is 1.

```yaml
development:
  adapter: mysql2
  connection_retries: 3
```

The database config also allows a `retry_deadline` to be configured. If a `retry_deadline` is configured,

> a query will _not_ be retried if a specified amount of time has elapsed since the query was first tried.

Reviewer: ```suggestion
an otherwise-retryable query will _not_ be retried if the specified time has elapsed while the query was first tried.
```
